packages/text-decorator/test/decorate-react.test.tsx

```diff
@@ -0,0 +1,251 @@
+import { decorateHtml, decorateReact, IDecorateHtmlOptions, IDecorateReactOptions } from "../src/text-decorator";
+import * as React from 'react';
+type ReactElement = React.ReactElement<any>;
+
+/**
+ * decorateReact tests
+ */
+describe("Test decorateReact (i.e. JSX/TSX)", () => {
+  const kNullOptions: IDecorateReactOptions = { words: [], replace: '' };
+  
+  it("handles primitive values", () => {
+    expect(decorateReact(null, kNullOptions)).toBe(null);
+    expect(decorateReact(undefined, kNullOptions)).toBe(undefined);
+    expect(decorateReact(0, kNullOptions)).toBe(0);
+    expect(decorateReact(1, kNullOptions)).toBe(1);
+    expect(decorateReact(false, kNullOptions)).toBe(false);
+    expect(decorateReact(true, kNullOptions)).toBe(true);
+  });
+
+  it("substitutes simple strings", () => {
+    const someOtherOptions: IDecorateReactOptions = { words: ['Some'], replace: 'Other' };
+    const textDataOptions: IDecorateReactOptions = { words: ['text'], replace: 'data' };
+    expect(decorateReact("Some Text", kNullOptions)).toBe("Some Text");
+    expect(decorateReact("Some Text", someOtherOptions)).toBe("Other Text");
+    expect(decorateReact("Text TEXT text", textDataOptions)).toBe("data data data");
+  });
+
+  it("ignores unknown objects", () => {
+    const empty = {};
+    expect(decorateReact(empty, kNullOptions)).toBe(empty);
+  });
+
+  it("handles arrays of simple values", () => {
+    const input = [undefined, null, 0, 1, false, true, "Some Text"];
+    const someMoreOptions: IDecorateReactOptions = { words: ['Some'], replace: 'More' };
+    const result = decorateReact(input, someMoreOptions);
+    expect(JSON.stringify(result)).toBe(JSON.stringify(input).replace("Some", "More"));
+  });
+
+  it("works with compiled JSX/TSX", () => {
+    const input = <div></div>;
+    const options = { words: ['foo'], replace: '' };
+    const result = decorateReact(input, options);
+    expect(JSON.stringify(result)).toBe(JSON.stringify(input));
+  });
+
+  it("works with compiled JSX/TSX", () => {
+    const input = <div>bar</div>;
+    const options = { words: ['foo'], replace: '' };
+    const result = decorateReact(input, options);
+    expect(JSON.stringify(result)).toBe(JSON.stringify(input));
+  });
+
+  it("works with compiled JSX/TSX", () => {
+    const input = <div>Some <span>Text</span></div>;
+    const result = decorateReact(input, kNullOptions);
+    expect(JSON.stringify(result)).toBe(JSON.stringify(input));
+  });
+
```

Comment:
These last few tests have the same title / description. (Though they test differing scenarios)